wallet/src/de/schildbach/wallet/ui/MainActivity.kt

```diff
@@ -281,11 +286,20 @@ class MainActivity : AbstractBindServiceActivity(), ActivityCompat.OnRequestPerm
     private fun setupBottomNavigation() {
         binding.bottomNavigation.itemIconTintList = null
         supportFragmentManager.addOnBackStackChangedListener {
-            if (supportFragmentManager.backStackEntryCount == 0) {
-                binding.bottomNavigation.selectedItemId = R.id.home
+            val lastFragment = supportFragmentManager.fragments.last()
+            val itemId = when(lastFragment::class) {
+                WalletFragment::class -> R.id.bottom_home
+                ContactsFragment::class,
+                UpgradeToEvolutionFragment::class -> R.id.contacts
+                PaymentsFragment::class -> R.id.payments
+                ExploreFragment::class -> R.id.discover
+                MoreFragment::class -> R.id.more
+                else -> binding.bottomNavigation.selectedItemId
             }
+
+            binding.bottomNavigation.menu.findItem(itemId).isChecked = true
```

Comment:
Instead of keeping the type of bottom navigation in the fragments themselves and having them check an appropriate bottom bar button, we can get the currently displayed fragment in the `MainActivity` and keep the `isChecked` assignment in here.